agent_controller/agent_controller.c

```diff
@@ -796,69 +998,288 @@ agent_controller_update_agents (agent_controller_connector_t conn,
 }
 
 /**
- * @brief Deletes a list of agents.
+ * @brief Updates scan agent config globally
  *
- * @param[in] conn Active connector
- * @param[in] agents List of agents to delete
+ * @param[in] cfg  Scan-agent configuration to serialize. Must not be NULL.
  *
- * @return RESP_CODE_OK (0) on success, RESP_CODE_ERR (-1) on failure
+ * @return Newly allocated, unformatted JSON string on success; NULL on failure
+ *         The caller owns the returned string and must free it with
+ *         g_free().
```

Comment:
```suggestion
 *         cJSON_free().
```
Minor, in case you're in the area again.